spring-boot-project/spring-boot-test/src/main/java/org/springframework/boot/test/context/SpringBootConfigurationFinder.java

```diff
@@ -33,20 +34,27 @@
  *
  * @author Phillip Webb
  */
-final class SpringBootConfigurationFinder {
+public final class SpringBootConfigurationFinder {
 
 	private static final Map<String, Class<?>> cache = Collections
 			.synchronizedMap(new Cache(40));
 
 	private final ClassPathScanningCandidateComponentProvider scanner;
 
-	SpringBootConfigurationFinder() {
+	public SpringBootConfigurationFinder() {
```

Comment:
We need Javadoc to clarify the intent compared to the second one you've asked. Also, I'd put it below the one you've created.